api/src/main/java/org/openmrs/CohortMembership.java

```diff
@@ -94,15 +94,15 @@ public void setPatientId(Integer patientId) {
 	}
 	
 	public Date getStartDate() {
-		return startDate;
+		return (Date) startDate.clone();
 	}
 	
 	public void setStartDate(Date startDate) {
-		this.startDate = startDate;
+		this.startDate = endDate != null ? (Date) startDate.clone() : null; 
```

Comment:
Also check [this](http://www.informit.com/articles/article.aspx?p=31551&seqNum=2), you should deep copy instead of clone in setters

Reply:
done!